api/create_hypertable.md

```diff
@@ -49,6 +49,14 @@ still work on the resulting hypertable.
  If you use `SELECT * FROM create_hypertable(...)` you get the return value formatted as a table with column headings.
 </highlight>
 
+<highlight type="note"> 
+The primary key for your PostgreSQL table must include
+all partitioning columns for your hypertable. Since all TimescaleDB hypertables
+are partitioned by time, your primary key must include the `time` column. If you
+specify a `partitioning_column`, your primary key must include that column as 
+well.
```

Comment:
One thing to note here and possibly clarify as it is a bit confusing at first read is that this does not work in the other direction, ie all the columns of your primary key do not need to be included in the partitioning columns, we really don't want people to think that or do that, it will lead to massive over-partitioning if we're not careful.

Maybe something like: 
"""
If you plan to have a `PRIMARY KEY` or `UNIQUE` constraint on your hypertable, it must include all columns that the hypertable is partitioned by. 

All TimescaleDB hypertables are partitioned by a time_column, therefore, this column must always be included in `PRIMARY KEY`s or `UNIQUE` constraints on a hypertable. If you specify a `partitioning_column`,  you must include that column in `PRIMARY KEY`s as well.

This does not work in the opposite direction, you may include columns that are not `partitioning_column`s or `time_column`s in a `PRIMARY KEY` or `UNIQUE` constraint. 

for further info see: 

And then we should split this out into its own explanation and have an example like: 

Text from above...
For instance, given a table such as 

```
CREATE TABLE ht_test(
ts TIMESTAMPTZ,
user_id BIGINT,
device_id BIGINT,
val1 FLOAT,
val2 FLOAT, 
PRIMARY KEY (device_id, ts));
```
You'll note that it has a `PRIMARY KEY` on `device_id, ts` - this means that at any timestamp `ts` you can only have one row per `device_id`.

This `PRIMARY KEY` limits the allowed partitioning schemes we can use, for instance, we could partition by `ts` only by using this `create_hypertable` call:

`SELECT * FROM create_hypertable('ht_test', 'ts');`
Here the `PRIMARY KEY` has two columns, `device_id` and `ts`, and we are partitioning by `ts` as that is our `time_column`, so this works, all columns we are partitioning by are in the `PRIMARY KEY`, and it includes another column, `device_id`. 

Alternatively, we could include `device_id` as a partitioning column as well:
`SELECT * FROM create_hypertable('ht_test', 'ts', 'partitioning_columns'=>'device_id');` 

Again, this works, as `device_id` and `ts` are in our `PRIMARY KEY` and are our partitioning columns. 

However, we cannot include `user_id` in our partitioning columns if it is not part of our `PRIMARY KEY`, so: 
`SELECT * FROM create_hypertable('ht_test', 'ts', 'partitioning_columns'=>'user_id');` 
will result in an error such as: 

```
 ERROR: cannot create a unique index without the column "<column_name>" (used in partitioning)
 ```

However, you could alter the PRIMARY KEY to include the `user_id` column and use any of the partitioning schemes above. 
"""

Reply:
I also think it would be useful to have a section on *WHY* this is, which I would happily write up or maybe work on a blog post about first if you'd prefer...

Reply:
I think "why" would be great! It might work better as a blog post, which we then take the highlights from for the docs. If you're interested in writing it, I'd love to see it.

Reply:
Honestly, I think we could add a paragraph (or two at most) at the beginning of the new Troubleshooting section that simply says, "Hypertables use PostgreSQL partitioning and a constraint on doing partitioning effectively is that any unique constraints must be a part of the partitioning keys. This is to prevent unnecessary work across partitions when dealing with lookups, Therefore..." - or whatever makes sense. If you can give us 3-4 sentences @davidkohn88 of what you think would be helpful as a starting place that would be much appreciated!